pkg/slices/slices.go

```diff
@@ -180,6 +180,19 @@ func Map[E any, O any](s []E, f func(E) O) []O {
 	return n
 }
 
+// MapErr runs f() over all elements in s and returns the result, short circuiting if there is an error.
+func MapErr[E any, O any](s []E, f func(E) (O, error)) ([]O, error) {
```

Comment:
we can use MapFilter instead

Reply:
then we don't know if there was an error or not and cannot handle it appropriately